### PR TITLE
Fix dolt diff --limit

### LINKED ISSUE #4078: diff limit flag ignored
No matter what `--limit` value you put, the entire diff will be printed

### PR DESCRIPTION
**Pull request by abmyii**
Closes #4078.

### CODE REVIEW

**Comment by timsehn** [comment]
Needs at least a single bats test that tests undefined limit, limit 0, and limit > 1.

**Comment by abmyii** [comment]
> Needs at least a single bats test that tests undefined limit, limit 0, and limit > 1.

Sure, will do.

**Comment by abmyii** [comment]
> Needs at least a single bats test that tests undefined limit, limit 0, and limit > 1.

Done. Might have gone a bit overboard though!

**Comment by abmyii** [comment]
Right now limit 0 does nothing (shows the entire diff). Would it be preferable if it showed no rows (empty tables)?
e.g.
```
+---+----+----+----+----+----+----+
|   | pk | c1 | c2 | c3 | c4 | c5 |
+---+----+----+----+----+----+----+
+---+----+----+----+----+----+----+
```

**Comment by timsehn** [comment]
Seems like empty is the right approach.

**Comment by timsehn** [comment]
@abmyii you want us to finish this off for you? Or will you write that one last test?

File: integration-tests/bats/diff.bats [hunk 1]
```diff
@@ -869,3 +869,122 @@ SQL
...
+    [[ "$output" =~ "$EXPECTED_TABLE2" ]] || false
+
+    run dolt diff --limit 0
+
+    [[ "$output" =~ "diff --dolt a/test b/test" ]] || false
+    [[ "$output" =~ "--- a/test @" ]] || false
+    [[ "$output" =~ "+++ b/test @" ]] || false
+    [[ "$output" =~ "diff --dolt a/test2 b/test2" ]] || false
+    [[ "$output" =~ "--- a/test2 @" ]] || false
+    [[ "$output" =~ "+++ b/test2 @" ]] || false
```
Comment (timsehn) [inline integration-tests/bats/diff.bats:989]:
```suggestion
    [[ "$output" =~ "+++ b/test2 @" ]] || false
    
    run dolt diff --limit
    [ "$status" -ne 0 ]
```

**Review by timsehn — CHANGES_REQUESTED** [review]

File: integration-tests/bats/diff.bats (hunk 1 above)
[the remark this answers is not in the document]
Comment (abmyii) [reply integration-tests/bats/diff.bats:989]: Ah, I missed that one - thanks!

**Review by timsehn — APPROVED** [review]
This looks good. @fulghum can you get it merged in. I think it needs `repo_format`

**Comment by fulghum** [comment]
Nice work @abmyii! Thanks for taking the time to contribute this fix. 🙏 

I believe the `Format PR` check doesn't run successfully in PRs from forks, but I see everything else is looking green, so I'll go ahead and merge this one in and we'll get it out in our next release.

**Pull request merged**